### PR TITLE
Make it clearer that edition functions are `>=`, not `==`

### PR DESCRIPTION
**Pull request by compiler-errors**
r? @Nilstrieb 

We could also perhaps derive `Ord` on `Edition` and use comparison operators.

### CODE REVIEW

**Comment by clarfonthey** [comment]
> We could also perhaps derive `Ord` on `Edition` and use comparison operators.

This seems way more reasonable to me.

**Comment by workingjubilee** [comment]
Nils is /afk
r? compiler

File: compiler/rustc_span/src/lib.rs [hunk 1]
```diff
@@ -707,24 +707,28 @@ impl Span {
...
 
+    /// Are we allowed to use features from the Rust 2018 edition?
     #[inline]
-    pub fn rust_2018(self) -> bool {
-        self.edition().rust_2018()
+    pub fn at_least_rust_2018(self) -> bool {
+        self.edition().at_least_rust_2018()
     }
 
+    /// Are we allowed to use features from the Rust 2018 edition?
```
Comment (oli-obk) [inline compiler/rustc_span/src/lib.rs:722]:
```suggestion
    /// Are we allowed to use features from the Rust 2021 edition?
```

File: compiler/rustc_span/src/lib.rs [hunk 2]
```diff
@@ -707,24 +707,28 @@ impl Span {
...
     #[inline]
-    pub fn rust_2021(self) -> bool {
-        self.edition().rust_2021()
+    pub fn at_least_rust_2021(self) -> bool {
+        self.edition().at_least_rust_2021()
     }
 
+    /// Are we allowed to use features from the Rust 2018 edition?
```
Comment (oli-obk) [inline compiler/rustc_span/src/lib.rs:728]:
```suggestion
    /// Are we allowed to use features from the Rust 2024 edition?
```

**Review by oli-obk — CHANGES_REQUESTED** [review]
r=me with doc copy pasta fixed

**Pull request merged**